app/models/mixins/ansible_authentications.rb

```diff
@@ -29,43 +29,43 @@ def ansible_secret_key=(key)
     auth.save!
   end
 
-  def ansible_rabbitmq_password
-    auth = authentication_type(ANSIBLE_RABBITMQ_TYPE)
-    auth.nil? ? nil : auth.password
+  def ansible_rabbitmq_authentication
+    authentication_type(ANSIBLE_RABBITMQ_TYPE)
   end
 
-  def ansible_rabbitmq_password=(password)
-    auth = authentication_for_type(ANSIBLE_RABBITMQ_TYPE, "Ansible Rabbitmq Password")
+  def set_ansible_rabbitmq_authentication(password, userid = "tower")
```

Comment:
@carbonin what do you think?

`def set_ansible_rabbitmq_authentication(:userid => "tower", :password)`

You should then be able to do set_ansible_rabbitmq_authentication(:password => "12345").